### ISSUE TITLE
Search not loaded because of sensible headers of hosting webserver (insiders)

### ISSUE THREAD

**Issue by camold** [squidfunk/mkdocs-material#3616]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When hosting the static website produced with "--no-directory-urls" on a webservice, the (offline?) search does not work. Even when the docs are generated with insiders.
It seems that the headers of the hosting service forbid loading of the search scripts - I wonder if there is a way around this.

### Expected behaviour

Search renders normally

### Actual behaviour

Search does not initialize because most likely the script is not loaded - see browser console output below.
![grafik](https://user-images.githubusercontent.com/2109454/155288270-2fcaa4db-1192-437a-a356-41e6543862b9.png)

### Steps to reproduce

Browser Warnings are as follows:
![grafik](https://user-images.githubusercontent.com/2109454/155287680-f9347c00-597c-4eb7-9c24-f902cc11d786.png)
![grafik](https://user-images.githubusercontent.com/2109454/155288128-adcc953f-98d1-432f-a63b-7a9b9c4913f6.png)

Headers of the hosting service seem to interfere with the search script. The referred link in the error is https://developer.mozilla.org/de/docs/Web/HTTP/Headers/X-Content-Type-Options

### Package versions

- Python: 3.9.7
- MkDocs: mkdocs, version 1.2.3 from /usr/local/lib/python3.9/dist-packages/mkdocs (Python 3.9)
- Material: 8.2.1+insiders.4.9.1

### Configuration

```yaml
Does not seem to depend on the config, as long as search is enabled. The problem seems to be that the headers of the hosting service interfere with the (offline?) search scripts.
```

### System information

- Operating system: Windows
- Browser: Firefox

**Comment by squidfunk** [comment]
Thanks for reporting. Could you please check if the issue also happens with Insiders 4.6.1?

In general, you probably realize that we cannot control the headers of external hosting services. For example, on GitHub Pages there's no problem. Thus it might be necessary to contact your hosting provider. The headers are actually wrong – they should be `application/javascript`, because the worker script is not a binary. AFAICT, this could a misconfiguration of your hosting provider.

**Comment by camold** [comment]
Yeah I think the same. Thanks for investigating anyways and making sure its the webserver's problem.
I was just wondering why we're having the trouble with the search only - all other javascripts like diagrams or the admonitions work fine (the browser won't complain). I guess because the search has to actually load some extra files (the search index?)?

**Comment by squidfunk** [comment]
Probably because search is off-loaded to a web worker.
